PictureApp.API/Controllers/AuthController.cs

```diff
@@ -1,40 +1,68 @@
 using System;
 using System.Security.Claims;
 using System.Threading.Tasks;
+using MediatR;
+using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.IdentityModel.Tokens;
 using PictureApp.API.Dtos;
+using PictureApp.API.Extensions.Exceptions;
 using PictureApp.API.Providers;
 using PictureApp.API.Services;
-using PictureApp.API.Extensions.Exceptions;
+using PictureApp.Messaging;
 
 namespace PictureApp.API.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
     public class AuthController : ControllerBase
     {
-        private readonly IAuthTokenProvider _jwtToken;
+        private readonly IAuthTokenProvider _authTokenProvider;
         private readonly IAuthService _authService;
-
-        public AuthController(IAuthTokenProvider jwtToken, IAuthService authService)
+        private readonly IMediator _mediator;
+        
+        public AuthController(IAuthTokenProvider authTokenProvider, IAuthService authService, IMediator mediator)
         {
-            _jwtToken = jwtToken ?? throw new ArgumentNullException(nameof(jwtToken));
+            _authTokenProvider = authTokenProvider ?? throw new ArgumentNullException(nameof(authTokenProvider));
             _authService = authService ?? throw new ArgumentNullException(nameof(authService));
+            _mediator = mediator ?? throw new ArgumentNullException(nameof(mediator));
         }
 
         [HttpPost("register")]
         public async Task<IActionResult> Register(UserForRegisterDto userForRegister)
         {
             userForRegister.Email = userForRegister.Email.ToLower();
-
+            
             if (await _authService.UserExists(userForRegister.Email))
             {
                 return BadRequest("User name already exists");
             }
 
             await Task.Run(() => _authService.Register(userForRegister));
             
-            return StatusCode(201);
+            await _mediator.Publish(new UserRegisteredNotificationEvent(userForRegister.Email));
+            
+            return StatusCode(StatusCodes.Status201Created);
+        }
+
+        [HttpPost("activate")]
+        public async Task<IActionResult> Activate(UserForRegisterActivateDto userForRegisterActivate)
+        {
+            try
+            {
+                await _authService.Activate(userForRegisterActivate.Token);
+            }
+            catch (Exception ex)
+            {
+                if (ex is SecurityTokenExpiredException || ex is EntityNotFoundException)
```

Comment:
is it necessary to Use Only Global Exception on catch block? Maybe we can merge SecurityTokenExpiredException and EntityNotFoundException to main catch block and remove if clausure and throw

Reply:
This approach is for catching specific exceptions. First of all, the global one is handled and thereafter Security* and Entiy* are taking into consideration. This was my purpose. What do You mean by merge them?